From 722ac823089706c3c4ea71386eafb4d7062fa7c9 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Fri, 11 Jul 2014 14:20:47 +0200
Subject: [PATCH 13/43] usb: Fix iovec memleak on combined-packet free

Message-id: <1405088470-24115-15-git-send-email-kraxel@redhat.com>
Patchwork-id: 59845
O-Subject: [RHEL-7.1 qemu-kvm PATCH 14/37] usb: Fix iovec memleak on combined-packet free
Bugzilla: 1075846
RH-Acked-by: Dr. David Alan Gilbert (git) <dgilbert@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

From: Hans de Goede <hdegoede@redhat.com>

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(cherry picked from commit 0ca6db4f3b3df5c4e5285a48a7709bdced5068de)
---
 hw/usb/combined-packet.c | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/usb/combined-packet.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/usb/combined-packet.c b/hw/usb/combined-packet.c
index 13f6602..ad77705 100644
--- a/hw/usb/combined-packet.c
+++ b/hw/usb/combined-packet.c
@@ -39,6 +39,7 @@ static void usb_combined_packet_remove(USBCombinedPacket *combined,
     p->combined = NULL;
     QTAILQ_REMOVE(&combined->packets, p, combined_entry);
     if (QTAILQ_EMPTY(&combined->packets)) {
+        qemu_iovec_destroy(&combined->iov);
         g_free(combined);
     }
 }
-- 
1.8.3.1

